From 8925760a59f984dae26c3e927e70d023be8408a6 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 14 Feb 2014 01:37:38 -0500
Subject: [PATCH] refpolicy: allow read rand/urand device and add block_suspend for syslog-ng

Allow syslog-ng to read /dev/random, /dev/urandom, also add block_suspend
capability for it. This will fix below avc denied issues:

    type=AVC msg=audit(1392343010.395:64): avc: denied
    { read } for pid=1654 comm="syslog-ng" name="urandom" dev="devtmpfs" \
    ino=5039 scontext=system_u:system_r:syslogd_t:s15:c0.c1023 tcontext= \
    system_u:object_r:urandom_device_t:s0 tclass=chr_file

    type=AVC msg=audit(1392343010.403:65): avc: denied \
    { read } for pid=1654 comm="syslog-ng" name="random" dev="devtmpfs" \
    ino=5038 scontext=system_u:system_r:syslogd_t:s15:c0.c1023 tcontext= \
    system_u: object_r:random_device_t:s0 tclass=chr_file

    type=AVC msg=audit(1392343012.571:82): avc: denied \
    { block_suspend } for pid=1654 comm="syslog-ng" capability=36 \
    scontext=system_u:system_r:syslogd_t:s15:c0.c1023 tcontext= \
    system_u:system_r:syslogd_t:s15:c0.c1023 tclass=capability2

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/logging.te | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
index f09e57e..7ecd9c1 100644
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -386,6 +386,12 @@ optional_policy(`
 # cjp: why net_admin!
 allow syslogd_t self:capability { chown dac_override fsetid net_admin setgid setuid sys_admin sys_nice sys_resource sys_tty_config };
 dontaudit syslogd_t self:capability { sys_ptrace };
+
+# allow read to /dev/random for syslog-ng
+dev_read_rand(syslogd_t)
+# block_suspend for syslog-ng
+allow syslogd_t self:capability2 block_suspend;
+
 # setpgid for metalog
 # setrlimit for syslog-ng
 # getsched for syslog-ng
-- 
2.7.4

